tests/Filter/CallbackFilterTest.php

```diff
@@ -108,4 +108,29 @@ public function testApplyMethod(): void
         $this->assertSame(['value' => 'myValue'], $builder->parameters);
         $this->assertTrue($filter->isActive());
     }
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @group legacy
+     */
+    public function testWrongCallbackReturnType(): void
+    {
+        $builder = new ProxyQuery(new QueryBuilder());
+
+        $filter = new CallbackFilter();
+        $filter->initialize('field_name', [
+            'callback' => static function ($builder, $alias, $field, $value) {
+                $builder->andWhere(sprintf('CUSTOM QUERY %s.%s', $alias, $field));
+                $builder->setParameter('value', $value);
+
+                return 1;
+            },
+        ]);
+
+        $filter->filter($builder, 'alias', 'field', 'myValue');
```

Comment:
Shouldn't we expect a deprecation in this call?

Reply:
Yes indeed, I forgot to add one.

Reply:
Great! Didn't see it before.